### PR TITLE
Adds Bitbucket Server (Stash) Pull Request plugin

### PR DESCRIPTION
**Pull request by asnodgrass**
Displays a list of open pull requests in for all repositories in a specified project on Bitbucket Server (née Stash) in which the user is a participant. Results are sorted by last update (ascending), and are prefixed with a character to indicate participation type: A for author, R for reviewer, and nothing for other (e.g. added a comment). Each entry is clickable and will take you to the pull request page.

Plugin Example

### CODE REVIEW

File: Dev/BitbucketServer/stash-pull-requests.2m.rb [hunk 1]
```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/env ruby
+
+require 'duration'
+require 'json'
+require 'rest-client'
+require 'pp'
+
+# rubocop:disable all
```
Comment (keithamus) [inline Dev/BitbucketServer/stash-pull-requests.2m.rb]: I feel like this is cheating :wink:.

Comment (tresni) [inline Dev/BitbucketServer/stash-pull-requests.2m.rb]: totally missed that, I don't really read comments other then metadata... I've reverted my merge.  Yeah, not sure how comfortable I am with that.

File: Dev/BitbucketServer/stash-pull-requests.2m.rb [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+# <bitbar.title>Bitbucket Server (Stash) Pull Requests</bitbar.title>
+# <bitbar.version>v1.0.0</bitbar.version>
+# <bitbar.author>Adam Snodgrass</bitbar.author>
+# <bitbar.author.github>asnodgrass</bitbar.author.github>
+# <bitbar.desc>Displays a list of open pull requests in for all repositories in a specified project on Bitbucket Server (aka Stash) in which the user is a participant. Results are sorted by last update (ascending), and are prefixed with a character to indicate participation type: A for author, R for reviewer, and nothing for other (e.g. added a comment). Each entry is clickable and will take you to the pull request page.</bitbar.desc>
+# <bitbar.image>http://i.imgur.com/9vp4D6e.png</bitbar.image>
+# <bitbar.dependencies>rest-client gem, ruby_duration gem</bitbar.dependencies>
+# rubocop:enable all
```
Comment (tresni) [inline Dev/BitbucketServer/stash-pull-requests.2m.rb]: Re-enabled here.  So it's just disabling them for the metadata block I think @keithamus

File: Dev/BitbucketServer/stash-pull-requests.2m.rb (hunk 1 above)
Comment (asnodgrass) [inline Dev/BitbucketServer/stash-pull-requests.2m.rb]: That's only for the block of metadata, not the whole file.

File: Dev/BitbucketServer/stash-pull-requests.2m.rb (hunk 2 above)
Comment (keithamus) [inline Dev/BitbucketServer/stash-pull-requests.2m.rb]: Oh yeah... my bad, didn't spot that!

**Comment by tresni** [comment]
Reverting my revert..  :shipit: (nothingtodohere)

**Pull request merged**